rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromReturnNewRector/Fixture/fallback_from_param_self.php.inc

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromReturnNewRector\Fixture;
+
+final class FallbackFromParamSelf
+{
+    public function action(self $obj)
+    {
+        if (rand(0, 1)) {
+            return new FallbackFromParamSelf();
+        }
+
+        return $obj;
+    }
+}
+
+?>
+-----
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromReturnNewRector\Fixture;
+
+final class FallbackFromParamSelf
+{
+    public function action(self $obj): \Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromReturnNewRector\Fixture\FallbackFromParamSelf
```

Comment:
Do we need another test for a `static` and/or `parent` typed param?

Reply:
static already error on the first place :)

- https://3v4l.org/cEZik
-

Reply:
multiple return already skipped at https://github.com/rectorphp/rector-src/blob/c956f862094345e72533d83e629938437bf29a43/rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromReturnNewRector.php#L185-L186